src/libcore/benches/slice.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use test::black_box;
+use test::Bencher;
+
+#[bench]
+fn binary_search(b: &mut Bencher) {
+    let mut v = Vec::new();
```

Comment:
`(0..999).collect::<Vec<_>>()` will be more efficient and somewhat easier to read (at least to me).

Reply:
Done.